src/Popover/Popover.js

```diff
@@ -65,7 +64,7 @@ class Popover extends React.PureComponent<Props, State> {
 
             {/* Note: Arrow */}
             <Overlay show={isOpen} resize target={target} {...position.arrow}>
-              <Arrow key="arrow" />
+              <StyledArrow key="arrow" />
```

Comment:
Looks like setting `pointer-event` on Arrow CANNOT avoid `onHide` getting called

When in `onHide`, `event.target` will be _container div of `StyledArrow`_, 
which is NEVER children of the dialog DOM.

REF: [storybook: Popover > in dialog](https://deploy-preview-125--mcs-ui.netlify.com/?selectedKind=Popover&selectedStory=in%20dialog&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybook%2Factions%2Factions-panel)